qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -772,6 +773,123 @@ def _add_common_creation_steps_to_queue(cls, md_template, obj_id,
+        ----------
+        md_template : DataFrame
+            The metadata template file contents indexed by sample ids
+        conn_handler : SQLConnectionHandler
+            The connection handler object connected to the DB
+        queue_name : str
+            The queue where the SQL statements will be added
+
+        Raises
+        ------
+        QiitaDBError
+            If no new samples or new columns are present in `md_template`
+        """
+        # Check if we are adding new samples
+        sample_ids = md_template.index.tolist()
+        sql = """SELECT sample_id FROM qiita.{0}
+                 WHERE {1}=%s""".format(self._table, self._id_column)
+        curr_samples = set(
+            s[0] for s in conn_handler.execute_fetchall(sql, (self._id,)))
+        existing_samples = curr_samples.intersection(sample_ids)
+        new_samples = set(sample_ids).difference(existing_samples)
+
+        # Check if we are adding new columns
+        table_name = self._table_name(self._id)
+        # Get the required columns from the DB
+        db_cols = sorted(get_table_cols(self._table, conn_handler))
+        # Remove the sample_id and _id_column columns
+        db_cols.remove('sample_id')
+        db_cols.remove(self._id_column)
+        # Get the columns from the dynamic table and do the union with the db
+        curr_cols = set(
+            get_table_cols(table_name, conn_handler)).union(db_cols)
+        headers = md_template.keys().tolist()
+        existing_cols = curr_cols.intersection(headers)
+        new_cols = set(headers).difference(existing_cols)
+
+        if not new_cols and not new_samples:
+            raise QiitaDBError(
+                "No new samples or new columns found in the template. If you "
+                "want to update existing values, you should use the 'update' "
+                "functionality.")
+
+        if new_cols:
+            # If we are adding new columns, add them first (simplifies code)
+            # Sorting the new columns to enforce an order
+            new_cols = sorted(new_cols)
+            datatypes = get_datatypes(md_template.ix[:, new_cols])
+            sql_cols = """INSERT INTO qiita.{0} ({1}, column_name, column_type)
+                          VALUES (%s, %s, %s)""".format(self._column_table,
+                                                        self._id_column)
+            sql_alter = """ALTER TABLE qiita.{0} ADD COLUMN {1} {2}"""
+            for category, dtype in zip(new_cols, datatypes):
+                conn_handler.add_to_queue(
+                    queue_name, sql_cols, (self._id, category, dtype))
+                conn_handler.add_to_queue(
+                    queue_name, sql_alter.format(table_name, category, dtype))
+
+            if existing_samples:
+                warnings.warn(
+                    "The new columns '%s' have been added to the existing "
+                    "samples '%s'. Any other value of these samples have been "
```

Comment:
I think the second sentence of this message is much more important than the first one, and users can miss it if they are updating a large number of columns. Also, I don't think the second sentence is very clear.